### PR TITLE
Release 0.6.0-rc2

### LINKED ISSUE #250: Update contracts to v0.6.0
Depends on https://github.com/confio/tgrade-contracts/issues/460

This includes quite a few renaming/api breaking changes. Please refer to docs and have a contract dev help with this

### PR DESCRIPTION
**Pull request by maurolacy**
Closes #250.

A summary of the changes, for reference:
### Changes

**tg4-engagement**

- `TG4Member::Weight` -> `TG4Member::Points`
- `TG4MemberResponse::Weight` -> `TG4MemberResponse::Points`
- `TG4TotalWeightResponse::Weight` -> `TG4TotalWeightResponse::Points`
- `ListMembersByWeight` -> `ListMembersByPoints`
- `Withdrawable/Distributed/UndistributedFunds` -> `Withdrawable/Distributed/UndistributedRewards`
- `FundsResponse` -> `RewardsResponse`, `Funds` -> `Rewards`

**tg4-stake**

- `TokensPerWeight` -> `TokensPerPoint`

**tgrade-valset**

- `MinWeight` -> `MinPoints`
- `RewardsCodeID` -> `ValidatorGroupCodeID`
- `RewardsContract` -> `ValidatorGroup`

### Currently unsupported changes

**tgrade-ap-voting**

- Contract not supported yet in tgrade.

**tgrade-oc-proposals**

- Added `ListVotesByVoter` query in contract.

**tg4-engagement**

- `distribute_funds` -> `distribute_rewards`. Not supported yet in tgrade.
- `withdraw_funds` -> `withdraw_rewards`. Not supported yet in tgrade.

**tgrade-community-pool**

- Added `ListVotesByVoter` query in contract.

**tgrade-validator-voting**

- Changed total_weight to total_points in ProposalResponse. Not supported yet in tgrade.
- Added `ListVotesByVoter` query in contract.
- Added `ListJailedValidators` query in contract.

### CODE REVIEW

File: x/poe/bootstrap.go [hunk 1]
```diff
@@ -304,12 +304,12 @@ func newEngagementInitMsg(gs types.GenesisState, adminAddr sdk.AccAddress) contr
 		PreAuthsHooks:    1,
 		PreAuthsSlashing: 1,
 		Denom:            gs.BondDenom,
-		Halflife:         uint64(gs.EngagmentContractConfig.Halflife.Seconds()),
+		Halflife:         uint64(gs.EngagementContractConfig.Halflife.Seconds()),
```
Comment (alpe) [inline x/poe/bootstrap.go:307]: 👍  good 👁️‍🗨️

File: x/poe/bootstrap.go [hunk 2]
```diff
@@ -321,7 +321,7 @@ func newStakeInitMsg(gs types.GenesisState, adminAddr sdk.AccAddress) contract.T
 		Admin:            adminAddr.String(),
 		Denom:            gs.BondDenom,
 		MinBond:          gs.StakeContractConfig.MinBond,
-		TokensPerWeight:  gs.StakeContractConfig.TokensPerWeight,
+		TokensPerPoint:   gs.StakeContractConfig.TokensPerWeight,
```
Comment (alpe) [inline x/poe/bootstrap.go:324]: `s/Weight/Point/` in the genesis naming

File: x/poe/contract/download_releases.sh [hunk 3]
```diff
@@ -9,19 +9,19 @@ fi
...
-GITHUB_API_TOKEN=${GITHUB_API_TOKEN:-"$(security find-generic-password -a "$USER" -s "github_api_key" -w)"}
-
 rm -f version.txt
 for contract in tg4_engagement tgrade_valset tg4_mixer tg4_stake tgrade_gov_reflect tgrade_community_pool tgrade_validator_voting; do
   echo "Download $contract from poe-contracts"
   asset_url="https://github.com/confio/poe-contracts/releases/download/${tag}/${contract}.wasm"
   rm -f "./${contract}.wasm"
   # download the artifact
-  echo $asset_url
+  echo "$asset_url"
```
Comment (alpe) [inline x/poe/contract/download_releases.sh:18]: 🥇

**Review by alpe — CHANGES_REQUESTED** [review]
Excellent work. I am very happy that you took this one and also fixed some typos. 💯 

There are only some minor comments and a [failing system test](https://app.circleci.com/pipelines/github/confio/tgrade/636/workflows/852e0af4-ad2e-4d5b-9879-397b6f0736e3/jobs/2412) that still checks for `weight`.

File: x/poe/keeper/msg_server_test.go [hunk 4]
```diff
@@ -155,7 +155,7 @@ func TestUpdateValidator(t *testing.T) {
 					m.Description = stakingtypes.NewDescription(
 						"otherMoniker",
 						"otherIdentity",
-						"https://otherWebsite",
+						"http://otherWebsite.com",
```
[the remark this answers is not in the document]
Comment (maurolacy) [reply x/poe/keeper/msg_server_test.go:158]: In introduced this for "diversity", but will change it.

**Review by alpe — APPROVED** [review]
Thanks for the updates! 💐

**Pull request merged**